src/bin/opensmt.cc

```diff
@@ -238,4 +210,80 @@ void catcher( int sig )
   }
 }
 
+
+void printHelp( )
+{
+    const char help_string[]
+        = "Usage: ./opensmt [OPTION] filename\n"
+          "where OPTION can be\n"
+          "  --help [-h]              print this help message and exits\n"
+          "  --version                prints opensmt version and exits\n"
+          "  -d                       executes dry run of opensmt\n"
+          "  -r <seed>                sets random seed to specific number\n"
+          "  -i                       produces interpolant in case of unsat\n"
+          "  -v                       verbose run of the solver\n"
+          "  -p                       for execution within a pipe\n"
+          "  --config=<filename>      use configuration file <filename>\n";
+    std::cerr << help_string;
+    exit( 0 );
+}
+
+SMTConfig parseCMDLine( int argc, char * argv[ ] )
+{
+    SMTConfig res;
+    int version = 0;
+
+    struct option long_options[] =
+        {
+            {"help", no_argument, nullptr, 'h'},
+            {"version", no_argument, &version, 1},
+            {"verbosity", no_argument, nullptr, 'v'},
+            {"dry-run", no_argument, nullptr, 'd'},
+            {"produce-interpolant", no_argument, nullptr, 'i'},
+            {"pipe", required_argument, nullptr, 'p'},
```

Comment:
I am pretty sure the pipe mode is independent of line editing. In fact, in pipe mode, line editing does not make sense.

Pipe mode was mainly used for incremental track in SMT-COMP, as far as I remember.
It feels like there should be a way to handle input in unified way, but I don't know what the problems there are.

Reply:
That was also my first impression, but after looking into the code, if `ENABLE_LINE_EDITING` is *not* defined, the code always calls `Interpret::interpPipe()` regardless of the parameter `-p`.

Reply:
I see, you are right. It is weird but **not** having the parameter leads to a different than the default behaviour if `ENABLE_LINE_EDITING` is defined...
Can this be quickly sorted out without touching `interpPipe` and `interpretInteractive`? I would keep that for separate PR.

Reply:
Hence I would keep this option local to `opensmt.cc` as before and not propagate it into `SMTConfig`.

Reply:
> Pipe mode was mainly used for incremental track in SMT-COMP, as far as I remember.

That's correct, and I think it still is used there?

> It feels like there should be a way to handle input in unified way, but I don't know what the problems there are.

Probably...  But it's not unusual to see a separate pipe mode?  There is something about the buffered input that required a lot of attention to get right.  The issue was related to not necessarily reading a parseable string because of how the buffering works.  The solution was to pass stuff to parser only once we have something where the parentheses match.  It was tricky to get right because of parentheses appearing in comments, and there's a whole state machine I had to write because of that.  I'm not sure if it's worth trying to generalize this, it doesn't sound like the most urgent thing now (even if at Certora we'll be using this mode, so maybe I'll need to come back with this...)

Reply:
The problem with it is that I need to extract it from the `parseCMDLineArgs`, and I would prefer this function to return purely `SMTConfig` to be honest...

But with this fixed return of `parseCMDLineArgs` parsing `-p` option outside of `SMTConfig` would be very painful

Reply:
> But it's not unusual to see a separate pipe mode?

Sorry, I didn't know that we always use the pipe reader (except for when using readline and friends).  I suppose it's working then and isn't a bottleneck :)

Reply:
I don't remeber the line editing API, this was written long time ago.  Maybe it can be improved, but it's probably rarely used (even if I like it!) and therefore not worth the time.

Reply:
> The problem with it is that I need to extract it from the `parseCMDLineArgs`, and I would prefer this function to return purely `SMTConfig` to be honest...
> 
> But with this fixed return of `parseCMDLineArgs` parsing `-p` option outside of `SMTConfig` would be very painful

Well it is not, it can be just stored in a separate variable. It would be better to deal with the whole command line stuff using a class, but sticking to the current state, it is sufficient to use an `inline bool` variable at global scope of `opensmt.cc`. This option is related only to parsing and does not affect the SMT solver at all, which is why I think we should not clog `SMTConfig` with another global option.

Reply:
I would do that only when it starts to be necessary. Here we discuss an option that is probably going to be removed in a next PR.